### ISSUE TITLE
MPL drawer support for complex custom instructions

### ISSUE THREAD

**Issue by 1ucian0** [Qiskit/qiskit-terra#3201]
The following circuit is being draw like wrongly:
```python
from qiskit.circuit import Gate, Parameter
qinst = QuantumRegister(1, 'q')
cinst = ClassicalRegister(1, 'c')

inst = QuantumCircuit(qinst, cinst, name='instruction').to_instruction()

qr = QuantumRegister(2, 'q')
cr = ClassicalRegister(2, 'c')
circuit = QuantumCircuit(qr, cr)
circuit.append(inst, [qr[0]], [cr[1]])
circuit.draw(output='mpl')
```
![image](https://user-images.githubusercontent.com/766693/66247245-9817ab80-e6e8-11e9-999f-4dbbd7c79e7b.png)

It should be similar to this:
Considering the following custom instruction:
```
        ┌─────────────┐            
q_0: |0>┤1            ├            
        │             │            
q_1: |0>┤             ├            
        │ instruction │            
 c_0: 0 ╡             ╞            
        │             │            
 c_1: 0 ╡0            ╞            
        └─────────────┘
```

Additionally, based on the consensus in #3073, the mpl drawer should support controlled/conditional custom instructions. The following is an example for conditional:
```
from qiskit.circuit import Gate, Parameter
qinst = QuantumRegister(1, 'q')
cinst = ClassicalRegister(1, 'c')

inst = QuantumCircuit(qinst, cinst, name='instruction').to_instruction()

qr = QuantumRegister(2, 'q')
cr = ClassicalRegister(2, 'c')
circuit = QuantumCircuit(qr, cr)
circuit.append(inst, [qr[0]], [cr[1]]).c_if(cr, 2)
circuit.draw(output='mpl')
```

Controlled is not yet supported (see #2862).

**Comment by yesodharan1892** [comment]
hi, 
i think i have the solution for that , can you assign it to me so that i can look into it.

**Comment by ajavadia** [comment]
@yesodharan1892 can you draw instances of `ControlledGate` in matplotlib's circuit visualizer? This is kind of high priority. Currently this is only doable in the text drawer (https://github.com/Qiskit/qiskit-terra/pull/3442)

**Comment by yesodharan1892** [comment]
from qiskit.circuit import Gate, Parameter
qr = QuantumRegister(2, 'q')
cr = ClassicalRegister(2, 'c')
inst = QuantumCircuit(qr, cr, name='instruction').to_instruction()
%matplotlib inline

circuit = QuantumCircuit(qr, cr)
circuit.append(inst, qr, cr)
circuit.draw(output='mpl')


with these instructions the mpl representation looks like this 
![1123](https://user-images.githubusercontent.com/57726155/72531177-f121de80-3896-11ea-91d1-31103e05fdfe.jpg)
 i made the inst QuantumCircuit to be a 2 qubit as initializing with a single qubit was creating the same error.

**Comment by yesodharan1892** [comment]
hi @ajavadia i have made some changes in the code given above and the mpl visualization is what as is expected, but when i look at the latex visualization it seems to be wrong.

from qiskit import *
from qiskit.circuit import Gate, Parameter
qr = QuantumRegister(2, 'q')
cr = ClassicalRegister(2, 'c')
inst = QuantumCircuit(qr, cr, name='instruction').to_instruction()
%matplotlib inline
circuit = QuantumCircuit(qr, cr)
circuit.append(inst, [qr[1], qr[0]],[cr[0],cr[1]])
circuit.draw(output='mpl')

![Untitled](https://user-images.githubusercontent.com/57726155/72533972-0fd6a400-389c-11ea-9efc-17762c285305.png)

![Untitled](https://user-images.githubusercontent.com/57726155/72533666-8921c700-389b-11ea-9225-3947b7f3d221.png)
this is the expectated visualization for q0 acting on c0 as it is shown below.

![Untitled](https://user-images.githubusercontent.com/57726155/72533816-cede8f80-389b-11ea-86ec-6491c74459df.png)

**Comment by 1ucian0** [comment]
> i look at the latex visualization it seems to be wrong.

you mean "text visualizer"?

**Comment by yesodharan1892** [comment]
yes exactly..

**Comment by 1ucian0** [comment]
Ping @yesodharan1892 . Any news?

**Comment by yesodharan1892** [comment]
Hi apologies as I've been offline due to some personal reasons. Seems the issue is addressed as per #6339
